crates/sui-indexer/src/apis/coin_api_v2.rs

```diff
@@ -1,28 +1,25 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-// TODO remove after the functions are implemented
-#![allow(unused_variables)]
-#![allow(dead_code)]
-
-use crate::store::PgIndexerStoreV2;
+use crate::indexer_reader::IndexerReader;
 use async_trait::async_trait;
 use jsonrpsee::core::RpcResult;
 use jsonrpsee::RpcModule;
-use sui_json_rpc::api::CoinReadApiServer;
+use sui_json_rpc::api::{cap_page_limit, CoinReadApiServer};
+use sui_json_rpc::coin_api::parse_to_type_tag;
 use sui_json_rpc::SuiRpcModule;
-use sui_json_rpc_types::{Balance, CoinPage, SuiCoinMetadata};
+use sui_json_rpc_types::{Balance, CoinPage, Page, SuiCoinMetadata};
 use sui_open_rpc::Module;
 use sui_types::balance::Supply;
 use sui_types::base_types::{ObjectID, SuiAddress};
 
 pub(crate) struct CoinReadApiV2 {
-    pg_store: PgIndexerStoreV2,
+    inner: IndexerReader,
 }
 
 impl CoinReadApiV2 {
-    pub fn new(pg_store: PgIndexerStoreV2) -> Self {
-        Self { pg_store }
+    pub fn new(inner: IndexerReader) -> Self {
+        Self { inner }
     }
 }
 
```

Comment:
nit: would some docs help here? For example, defaulting to gas coin type (while probably pretty common), is useful to know. 
The reason I am asking is that in the Rust SDK I used this functions and some doc comments would have helped a lot.

Reply:
The comment can be found in the trait [here](https://github.com/MystenLabs/sui/blob/main/crates/sui-json-rpc/src/api/coin.rs#L20)
and that's where the [api docs](https://docs.sui.io/sui-jsonrpc#suix_getCoins) pull from

Reply:
Awesome!